src/components/charts/bar/Bar.js

```diff
@@ -73,6 +75,7 @@ const Bar = ({
         })
     }
 
+    const getLabelsFormat = getLabelGenerator(label, labelFormat)
```

Comment:
This function should be issued by the *enhancer* (`enhance.js`) as if we re-compute it each time we render the bar, it will force React to render each labels because the reference will change each time, if you compute it inside the *enhancer*, it will be re-computed only if `label`or `labelFormat` changed, so we keep the shallow compare ability.

```javascript
// /src/components/charts/bar/enhance.js
export default Component =>
    compose(
        // …
        withPropsOnChange(['label', 'labelFormat'], ({ label, labelFormat }) => ({
            getLabel: getLabelGenerator(label, labelFormat),
        })),
        pure
    )(Component)
```

Reply:
Also, it will make the `getLabel()` function available for `BarCanvas` component (but you'll have to add support for it inside this component).

Reply:
Sorry, in fact I didn't added labels support in `BarCanvas` :/ But it will be easier for me to do so with this approach :)